File: frontend/src/types/fiscal-year.ts
export interface FiscalYear {
  id: string;
  name: string;
  start_date: string;
  end_date: string;
  status: 'open' | 'closed';
  created_at: string;
  updated_at: string;
}

export interface FiscalPeriod {
  id: string;
  fiscal_year_id: string;
  name: string;
  start_date: string;
  end_date: string;
  period_number: number;
  status: 'open' | 'closed';
}

export interface CreateFiscalYear {
  name: string;
  start_date: string;
  end_date: string;
}

export interface UpdateFiscalYear {
  name?: string;
  start_date?: string;
  end_date?: string;
}

File: backend/crates/konto-db/src/entities/fiscal_year.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "fiscal_years")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    pub name: String,
    pub start_date: chrono::NaiveDate,
    pub end_date: chrono::NaiveDate,
    pub status: String,
    pub created_at: chrono::NaiveDateTime,
    pub updated_at: chrono::NaiveDateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::fiscal_period::Entity")]
    FiscalPeriods,
}

impl Related<super::fiscal_period::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::FiscalPeriods.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
